src/frontend/src/pages/GanttPage/GanttChartComponents/GanttTaskBar/GanttTaskBarDisplay.tsx

```diff
@@ -0,0 +1,170 @@
+    gridRow: 1,
+    zIndex: 1
+  };
+
+  const ganttTaskBarDetailsBoxStyles: CSSProperties = {
+    gridRow: 1,
+    zIndex: 3,
+    gridColumnStart: getStartCol(task.start),
+    gridColumnEnd: getEndCol(task.end),
+    display: 'flex',
+    alignItems: 'center',
+    marginTop: isProject ? '-10px' : undefined,
+    marginBottom: isProject ? '-10px' : undefined,
+    cursor: 'pointer',
+    width: isProject ? 'fit-content' : '100%'
+  };
+
+  const ganttTaskBarWorkPackageOverlayStyles = (child: GanttTask): CSSProperties => {
+    return {
+      gridColumnStart: getStartCol(child.start),
+      gridColumnEnd: getEndCol(child.end),
+      height: '2rem',
+      border: `1px solid ${theme.palette.divider}`,
+      borderRadius: '0.25rem',
+      backgroundColor: child.styles ? child.styles.backgroundColor : grey[700],
+      cursor: 'pointer',
+      gridRow: 1,
+      zIndex: 2
+    };
+  };
+
+  const highlightedChangeBoxStyles = (highlightedChange: RequestEventChange): CSSProperties => {
+    return {
+      paddingTop: '2px',
+      paddingLeft: '5px',
+      gridColumnStart: days.findIndex((day) => dateToString(day) === dateToString(highlightedChange.newStart)) + 1,
+      gridColumnEnd:
+        days.findIndex((day) => dateToString(day) === dateToString(highlightedChange.newEnd)) === -1
+          ? days.length + 1
+          : days.findIndex((day) => dateToString(day) === dateToString(highlightedChange.newEnd)) + 2,
+      height: '2rem',
+      border: `1px solid ${theme.palette.text.primary}`,
+      borderRadius: '0.25rem',
+      backgroundColor: '#ef4345',
+      cursor: 'pointer',
+      gridRow: 1,
+      zIndex: 6
+    };
+  };
+
+  return (
+    <Box style={ganttTaskBarContainerStyles()}>
+      <Box sx={ganttTaskBarBackgroundStyles(days.length)}>
+        <div
+          style={ganttTaskBarHoverDetectionBoxStyles}
+          onMouseOver={(e) => handleOnMouseOver(e, task)}
+          onMouseLeave={handleOnMouseLeave}
+          onClick={() => history.push(`${`${routes.PROJECTS}/${task.id}`}`)}
+        >
+          <Box sx={webKitBoxContainerStyles()}>
+            <Box sx={webKitBoxStyles()} />
+          </Box>
+        </div>
+        <div
+          style={ganttTaskBarDetailsBoxStyles}
+          onMouseOver={(e) => handleOnMouseOver(e, task)}
+          onMouseLeave={handleOnMouseLeave}
+          onClick={!isProject ? () => history.push(`${`${routes.PROJECTS}/${task.id}`}`) : undefined}
+        >
+          {isProject && (
+            <IconButton onClick={onWorkPackageToggle} sx={{ marginRight: '-15px', marginLeft: '-5px' }}>
+              {showWorkPackages ? <ArrowDropDownIcon fontSize="large" /> : <ArrowRightIcon fontSize="large" />}
+            </IconButton>
+          )}
+          <Typography variant="body1" sx={taskNameContainerStyles(task)} onClick={onWorkPackageToggle}>
+            {task.name}
+          </Typography>
+        </div>
+        {!highlightedChange &&
+          isProject &&
```

Comment:
combine these two checks into a "isCreateProjectChange" or smthn, assuming that's what this is

Reply:
I think i can get rid of the !highlightedChange